frappe/model/base_document.py

```diff
@@ -12,11 +12,9 @@
 from frappe.model.utils.link_count import notify_link_count
 from frappe.modules import load_doctype_module
 from frappe.model import display_fieldtypes
-from frappe.utils.password import get_decrypted_password, set_encrypted_password
 from frappe.utils import (cint, flt, now, cstr, strip_html,
 	sanitize_html, sanitize_email, cast_fieldtype)
 from frappe.utils.html_utils import unescape_html
-from bs4 import BeautifulSoup
```

Comment:
How I see is that where ever we go for lazy imports(function level imports), these are special cases like external packages where we do not really have control over.

Incase of internal code, can we organize the code in such a way that  we do not need function level imports (Yes, this is not in this PR scope).